enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,317 @@
+    constraints on every other Azure resource required by Openshift as well.
+3. A tag value can have a maximum of 256 characters.
+4. A resource, resource-group or subscription, user can configure a maximum of 5 tags
+   through Openshift. 
+    - Azure supports a maximum of 50 tags except for Automation, Content Delivery Network,
+    DNS resources which can have a maximum of 15 tags, hence restricting the number of 
+    user defined tags to 10 and 5 for Openshift's internal use, for all the resources 
+    created by Openshift.
+
+All in-cluster operators that create Azure resources (Cluster Infrastructure ,Storage ,Node ,NetworkEdge , Internal Registry ,CCO) will apply these tags during resource creation.
+
+The userTags field is intended to be set at install time and is considered immutable. 
+Components that respect this field must only ever add tags that they retrieve from this 
+field to cloud resources, they must never remove tags from the existing underlying cloud 
+resource even if the tags are removed from this field(despite it being immutable).
+
+If the userTags field is changed post-install, there is no guarantee about how an 
+in-cluster operator will respond to the change. Some operators may reconcile the 
+change and change tags on the Azure resource. Some operators may ignore the change. 
+However, if tags are removed from userTags, the tag will not be removed from the 
+Azure resource.
+
+### Workflow Description
+
+- An Openshift administrator requests to add required tags to all Azure resources 
+  created by Openshift by adding it in `.platform.azure.userTags`
+- openshift installer validates the tags defined in `.platform.azure.userTags` and 
+  adds these tags to all resources created during installation and also updates 
+  `.status.platformStatus.azure.resourceTags` of the `infrastructure.config.openshift.io`
+- In cluster operators refers `.status.platformStatus.azure.resourceTags` of the 
+  `infrastructure.config.openshift.io` to add tags to resources created later.
+
+#### Variation [optional]
+
+### API Extensions
+Enhancement requires below modifications to the mentioned CRDs
+- Add `userTags` field to `platform.azure` of the `installconfigs.install.openshift.io`
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: installconfigs.install.openshift.io
+spec:
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        properties:
+          platform:
+            properties:
+              azure:
+                properties:
+                  userTags:
+                    additionalProperties:
+                      type: string
+                    description: UserTags additional keys and values that the installer
+                      will add as tags to all resources that it creates. Resources
+                      created by the cluster itself may not include these tags.
+                  type: object
```

Comment:
not sure if I'm reading this right, and which type is referring to which, but i would expect `userTags` to be the equivalent of a map in go

Reply:
Yes, it will be of type `map[string]string`

Reply:
userTags in installconfig CR is defined as `map[string]string` type and the resourceTags in infrastructure CR is defined as slice of struct containing key and value as members of type string.